cirq-core/cirq/transformers/target_gatesets/cz_gateset.py

```diff
@@ -25,23 +25,53 @@
 
 
 class CZTargetGateset(compilation_target_gateset.TwoQubitCompilationTargetGateset):
-    """Target gateset containing CZ + single qubit rotations + Measurement gates."""
+    """Target gateset accepting CZ + single qubit rotations + measurement gates.
 
-    def __init__(self, *, atol: float = 1e-8, allow_partial_czs: bool = False) -> None:
+    By default, `cirq.CZTargetGateset` will accept and compile unknown gates to
+    the following universal target gateset:
+    - `cirq.CZ` / `cirq.CZPowGate`: The two qubit entangling gate.
+    - `cirq.PhasedXZGate`: Single qubit rotations.
+    - `cirq.MeasurementGate`: Measurements.
+
+    Optionally, users can also specify additional gates / gate families which should
+    be accepted by this gateset via the `additional_gates` argument.
+
+    When compiling a circuit, any unknown gate, i.e. a gate which is not accepted by
+    this gateset, will be compiled to the default gateset (i.e. `cirq.CZ`/`cirq.CZPowGate`,
+    `cirq.PhasedXZGate`, `cirq.MeasurementGate`).
+    """
+
+    def __init__(
+        self,
+        *,
+        atol: float = 1e-8,
+        allow_partial_czs: bool = False,
+        additional_gates: Sequence[Union[Type['cirq.Gate'], 'cirq.Gate', 'cirq.GateFamily']] = (
+            ops.GlobalPhaseGate,
```

Comment:
Another option is to leave out GlobalPhaseGate altogether and add in the docstring that a user most likely will want to include this. One way to think about the core gate set (PhasedXZ etc) is the set of gates the target gateset can compile to. If that's the abtraction, then GlobalPhaseGate doesn't belong there. But I don't have a strong preference among these options :) including the option of keeping it as is.

Reply:
The reason I wanted to include `GlobalPhaseGate` by default, instead of excluding it, is that we now preserve the global phase in all non-analytical decompositions, often by adding an explicit instance of `cirq.GlobalPhaseGate` to the decomposition (see https://github.com/quantumlib/Cirq/pull/5420). 

So, for example, if you try to compile a 3 qubit `cirq.CCZPowGate` gate using these target gatesets, we will fallback on the `cirq.decompose` protocol while expanding the composite gates (step-1 of preprocess transformers); the default expansion will add a GlobalPhaseGate to the decomposition. If the target gateset doesn't accept a global phase gate, it will raise an error. More often than not, we don't want to raise an error in this case.

For example, in most cases, even if a vendor backend doesn't support serializing global phase gates, they'd actually want to accept circuits containing global phase gates and ignore the global phase during serialization, since it doesn't have any measurable impact on the overall circuit execution.

Reply:
Got it, thanks for the explanation! Will move GlobalPhaseGate into the core gateset.

Reply:
Btw the files still need to be updated to include empty lists for `additional_gates` because the JSON serialization test verifies whether `cirq.to_json(x)` matches the content in the .json file, similar to what happened [here](https://github.com/quantumlib/Cirq/pull/5101/files#diff-f11b21ad177fcfdbb523ce4c2f5de84fe4f6b087f84d4854c21c81708c7e36b6)